platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```
@@ -0,0 +1,144 @@
+            A string containing the description of the component
+        """
+        return "BIOS - Basic Input/Output System"
+
+
+    def get_firmware_version(self):
+        """
+        Retrieves the firmware version of the component
+
+        Returns:
+            A string containing the firmware version of the component
+
+        BIOS version is retrieved via command 'dmidecode -t 11'
+        which should return result in the following convention
+            # dmidecode 3.0
+            Getting SMBIOS data from sysfs.
+            SMBIOS 2.7 present.
+
+            Handle 0x0022, DMI type 11, 5 bytes
+            OEM Strings
+                    String 1:*0ABZS017_02.02.002*
+                    String 2: To Be Filled By O.E.M.
+
+        By using regular expression 'OEM[\s]*Strings\n[\s]*String[\s]*1:[\s]*([0-9a-zA-Z_\.]*)'
+        we can extrace the version string which is marked with * in the above context
+        """
+        bios_ver_str = self._get_command_result(BIOS_QUERY_VERSION_COMMAND)
+        try:
+            m = re.search('OEM[\s]*Strings\n[\s]*String[\s]*1:[\s]*([0-9a-zA-Z_\.]*)', bios_ver_str)
+            result = m.group(1)
+        except:
+            result = ''
+
+        return result
+
+
+class ComponentCPLD(Component):
+    def __init__(self):
+        self.name = COMPONENT_CPLD
+
+
+    def get_description(self):
+        """
+        Retrieves the description of the component
+
+        Returns:
+            A string containing the description of the component
+        """
+        return "CPLD - includes all CPLDs in the switch"
+
+
+    def get_firmware_version(self):
+        """
+        Retrieves the firmware version of the component
+
+        Returns:
+            A string containing the firmware version of the component
+        """
+        cpld_version_file_list = glob(CPLD_VERSION_FILE_PATTERN)
+        cpld_version = ''
+        if cpld_version_file_list is not None and not cpld_version_file_list == []:
```

@stephenxs in case if cpld_version_file_list is empty we might need to throw the exception.

---

fixed.

---

Maybe something like this will be better:
```
if cpld_version_file_list is not None and not cpld_version_file_list.empty():
```

---

According to PEP8, the recommended practice is:

    if cpld_version_file_list is not None and cpld_version_file_list:

We should go this way.